src/app/snapshot/store/snapshot-page.selectors.ts

```diff
@@ -1,11 +1,167 @@
 import { createFeatureSelector, createSelector } from '@ngrx/store';
+import * as moment from 'moment';
+import * as volumeSnapshotsSelectors from '../../reducers/snapshots/redux/snapshot.reducers';
+import * as vmSelectors from '../../reducers/vm/redux/vm.reducers';
+import { vmSnapshotsSelectors } from '../../root-store/server-data/vm-snapshots';
+import { getSnapshotDescription, Snapshot } from '../../shared/models';
+import { getSnapshotType, VmSnapshot } from '../../shared/models/vm-snapshot.model';
+import { SnapshotType } from '../../shared/types';
+import { VirtualMachine, VmState } from '../../vm';
 import { Filters } from '../models/filters.model';
+import { VmSnapshotSidebarViewModel } from '../models/vm-snapshot-sidebar.view-model';
+import { VmSnapshotViewModel } from '../models/vm-snapshot.view-model';
 import { SnapshotPageState, snapshotPageStoreName } from './snapshot-page.reducer';
 
 const getSnapshotPageState = createFeatureSelector<SnapshotPageState>(snapshotPageStoreName);
 
+const getSelectedId = createSelector(getSnapshotPageState, state => state.selectedId);
+
 export const getViewMode = createSelector(getSnapshotPageState, state => state.viewMode);
 
 export const getFilters = createSelector(getSnapshotPageState, (state): Filters => state.filters);
 
 export const getGroupings = createSelector(getSnapshotPageState, state => state.groupings);
+
+export const getSelectedSnapshot = createSelector(
+  getSelectedId,
+  vmSnapshotsSelectors.selectAll,
+  volumeSnapshotsSelectors.selectAll,
+  (selectedId, vmSnapsots, volumeSnapshots): Snapshot | VmSnapshot => {
+    const volumeSnapshot = volumeSnapshots.find(snap => snap.id === selectedId);
+    if (volumeSnapshot != null) {
+      return volumeSnapshot;
+    }
+
+    return vmSnapsots.find(snap => snap.id === selectedId);
+  },
+);
+
+export const getSelectedSnapshotForSidebar = createSelector(
+  getSelectedSnapshot,
+  vmSelectors.selectEntities,
+  (snapshot, vmEntities): Snapshot | VmSnapshotSidebarViewModel => {
+    if (getSnapshotType(snapshot) === SnapshotType.VM) {
+      const vmSnapshot: VmSnapshot = snapshot as VmSnapshot;
+      const viewModel: VmSnapshotSidebarViewModel = {
+        id: vmSnapshot.id,
+        state: vmSnapshot.state,
+        name: vmSnapshot.displayname,
+        description: vmSnapshot.description,
+        vmId: vmSnapshot.virtualmachineid,
+        vmName: vmEntities[vmSnapshot.virtualmachineid].displayname,
+        vmState: vmEntities[vmSnapshot.virtualmachineid].state,
+        created: moment(vmSnapshot.created).toDate(),
+        type: vmSnapshot.type,
+        current: vmSnapshot.current,
+        parentName: vmSnapshot.parentName,
+      };
+      return viewModel;
+    }
+
+    return snapshot as Snapshot;
```

Comment:
Why do you only create ViewModel for VM snapshots? Volume snapshots have the `created` field as well.